driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/MapReducePublisherImpl.java

```diff
@@ -241,8 +244,13 @@ public String getCommandName() {
         }
 
         @Override
-        public void executeAsync(final AsyncReadBinding binding, final SingleResultCallback<AsyncBatchCursor<T>> callback) {
-            operation.executeAsync(binding, callback::onResult);
+        public BatchCursor<T> execute(final ReadBinding binding, final OperationContext operationContext) {
```

Comment:
No need to explicitly add this as it should be in the ReadOperationCursorAsyncOnly - lets update the default method there instead.